app/actions/handlers.py

```
@@ -138,113 +139,126 @@ async def action_pull_events(integration, action_config: PullEventsConfig):
+        # Send transformed data to Sensors API V2
+        try:
+            for i, batch in enumerate(generate_batches(transformed_data, 200)):
+                logger.info(f'Sending observations batch #{i}: {len(batch)} observations. AOI: {action_config.aoi}')
+                response = await gundi_tools.send_events_to_gundi(
+                    events=batch,
+                    integration_id=integration.id
+                )
+
+                if response:
+                    all_responses.extend(response)
+                    result["events_processed"] += len(response)
+                    # Send images as attachments (if available)
+                    await process_attachments(batch, response, integration)
+                    # Process events to patch
+            await save_events_state(all_responses, action_config.events, integration)
+        except (httpx.ConnectTimeout, httpx.ReadTimeout) as e:
+            msg = (f'Timeout exception. AOI: {action_config.aoi}. Integration: {str(integration.id)}. '
+                   f'Exception: {e}, Type: {str(type(e))}, Request: {str(e.request)}')
+            logger.exception(
+                msg,
+                extra={
+                    'needs_attention': True,
+                    'integration_id': str(integration.id),
+                    "aoi": action_config.aoi,
+                    'action_id': "pull_events"
+                }
+            )
+            raise e
+        except httpx.HTTPError as e:
+            msg = f'Sensors API returned error. AOI: {action_config.aoi}. Integration: {str(integration.id)}. Exception: {e}'
+            logger.exception(
+                msg,
+                extra={
+                    'needs_attention': True,
+                    'integration_id': str(integration.id),
+                    "aoi": action_config.aoi,
+                    'action_id': "pull_events"
+                }
+            )
+            result["details"]["error"] = msg
```

This may hide errors. Now that we have a sub-action per AOI I think it's safe to remove this error handling, let the execution of a sub-action fail, and the framework will log the error in activity logs.